server/sound-track/src/main/java/soundtrack/controllers/AuthController.java

```diff
@@ -38,7 +38,10 @@ public AuthController(UserService userService, AuthenticationManager authenticat
         try {
             Authentication authentication = authenticationManager.authenticate(authToken); //This is where the error gets thrown
             if (authentication.isAuthenticated()) {
-                User user = (User)authentication.getPrincipal();
+                org.springframework.security.core.userdetails.User userDetails =
+                        (org.springframework.security.core.userdetails.User)authentication.getPrincipal();
+
+                User user = userService.findByEmail(userDetails.getUsername());
```

Comment:
The `User` that comes back from the `getPrincipal()` method is the Spring Security `User` type. For now, the simplest thing to do in order to have all of the user information available for the token, is to use the server to retrieve the user. You can avoid this by creating a custom `User` class that either extends the Spring Security `User` class or implements the `UserDetails` interface. If you're interested in doing this, we can talk about it more later, if that'd help.